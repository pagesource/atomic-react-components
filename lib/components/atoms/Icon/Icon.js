// @flow
/**
 *
 * Icon
 *
 */
import React from 'react';
import type { ComponentType } from 'react';
import styled from 'styled-components';
import classnames from 'classnames';

import styles from './Icon.style';
import type { Props } from './types';

const Icon = ({ className, width, title, height, children, fill }: Props) => {
  const childrenNode = React.Children.map(children, (child) => {
    return React.cloneElement(child, {
      width: width || '25px',
      fill: fill || '#000',
      height: height || '25px',
      display: 'inline-flex',
    });
  });
  return (
    <span role="presentation" aria-label={title} className={classnames('icon-wrapper', className)}>
      {childrenNode}
    </span>
  );
};

const StyledIcon: ComponentType<Props> = styled(Icon)`
  ${styles};
`;
export default StyledIcon;

export { Icon as IconVanilla };
