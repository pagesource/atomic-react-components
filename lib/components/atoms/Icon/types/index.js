// @flow

export type Props = {
  className?: string,
  id: string,
  width?: string,
  height?: string,
  title: string,
};
