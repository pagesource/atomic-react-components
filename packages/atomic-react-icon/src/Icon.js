/**
 *
 * Icon
 *
 */
import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import classnames from 'classnames';

import styles from './Icon.style';

const Icon = ({ className, width, title, height, children, fill }) => {
  const childrenNode = React.Children.map(children, (child) => {
    return React.cloneElement(child, {
      width: width || '25px',
      fill: fill || '#000',
      height: height || '25px',
      display: 'inline-flex',
    });
  });
  return (
    <span role="presentation" aria-label={title} className={classnames('icon-wrapper', className)}>
      {childrenNode}
    </span>
  );
};

const StyledIcon = styled(Icon)`
  ${styles};
`;

Icon.propTypes = {
  className: PropTypes.string,
  width: PropTypes.string,
  height: PropTypes.string,
  title: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
  fill: PropTypes.string,
};

Icon.defaultProps = {
  className: '',
  width: '',
  height: '',
  fill: '',
};

export default StyledIcon;

export { Icon as IconVanilla };
