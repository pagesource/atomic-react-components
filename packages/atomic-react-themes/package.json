{
  "name": "@xt-pagesource/atomic-react-themes",
  "version": "1.0.0",
  "main": "lib/index.js",
  "module": "src/index.js",
  "dependencies": {
    "styled-components": "^5.0.1"
  },
  "license": "MIT"
}
