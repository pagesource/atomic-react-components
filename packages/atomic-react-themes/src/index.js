import { createGlobalStyle } from 'styled-components';
import globalStyles from './globalStyles';
import theme from './themes/base';

const globalStyleObj = createGlobalStyle`
  ${globalStyles}
`;

export { theme };
export default globalStyleObj;
