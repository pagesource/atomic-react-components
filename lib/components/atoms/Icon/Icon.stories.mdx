import { Meta, Story, Preview, Props } from '@storybook/addon-docs/blocks';
import  Icon, {IconVanilla}  from './index.js';
import { ReactComponent as IconClose } from '../../../static/images/close.svg';
import defaultConfig from './Icon.mock';

<Meta title="Atoms/Icons" component={IconVanilla} />

### Icon
'IconClose' - import { ReactComponent as IconClose } from 'svg img path';

<Preview>
  <Story name="Close Icon">
      <IconVanilla  {...defaultConfig} fill="#000" title="close icon"  width="30px" height="30px">
        <IconClose />
       </IconVanilla>
  </Story>
</Preview>

