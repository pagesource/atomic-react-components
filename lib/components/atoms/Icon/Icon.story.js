import React from 'react';
import { storiesOf } from '@storybook/react';
import { ReactComponent as IconClose } from '../../../static/images/close.svg';
// Import Styled Component to showcase variations
import Icon from './Icon';

storiesOf('Atoms/Icon', module)
  .addParameters({ jest: ['Icon', 'IconVanilla'] })
  .add('Knobs', () => (
    <Icon title="close icon" width="30px" height="30px">
      <IconClose />
    </Icon>
  ));
