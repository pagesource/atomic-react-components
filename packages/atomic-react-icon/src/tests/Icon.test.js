import React from 'react';
import { shallow } from 'enzyme';
import { theme } from '@xt-pagesource/atomic-react-themes';

import Icon from '../Icon';

const IconClose = theme.icons.close;

describe('<Icon />', () => {
  test('should render correctly', () => {
    const IconComponentWrapper = shallow(
      <Icon title="close icon" width="30px" height="30px">
        <IconClose />
      </Icon>
    );
    expect(IconComponentWrapper).toMatchSnapshot();
  });
});
