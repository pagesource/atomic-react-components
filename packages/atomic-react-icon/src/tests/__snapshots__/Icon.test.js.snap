// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> should render correctly 1`] = `
<Icon
  className="Icon__StyledIcon-sc-3g6lof-0 NVSQO"
  fill=""
  height="30px"
  title="close icon"
  width="30px"
>
  <displayName
    0="d"
    1="e"
    10="p"
    11="s"
    2="f"
    3="a"
    4="u"
    5="l"
    6="t"
    7="P"
    8="r"
    9="o"
  />
</Icon>
`;
